src/js/angular/repositories/ontop-repo-constants.js

```diff
@@ -0,0 +1,240 @@
+export const ONTOP_TYPE = 'ontop';
+export const GENERIC_DRIVER_TYPE = 'generic';
+export const PROPERTIES_FILE = 'propertiesFile';
+export const ONTOP_REPO_PARAM_LABELS = {propertiesFile: 'JDBC properties', obdaFile: 'OBDA or R2RML', owlFile: 'ontology file', constraintFile: 'constraint file'};
+export const ONTOP_REPO_PARAMS = _.keys(ONTOP_REPO_PARAM_LABELS);
+export const REQUIRED_ONTOP_REPO_PARAMS = [PROPERTIES_FILE, 'obdaFile'];
+export const SUPPORTED_DRIVER_LABELS = {hostName: 'Hostname', port: "Port", databaseName: 'Database name',
+                                        userName: 'Username', password: 'Password', driverClass: 'Driver class', url: 'URL'};
+export const PROPERTIES_FILE_PARAMS = _.keys(SUPPORTED_DRIVER_LABELS);
+export const REQUIRED_PROPERTIES_FIELD_PARAMS = ['hostName', 'databaseName', 'userName'];
+
+export const DEFAULT_SELECTED_DRIVER = {
+        driverType: "generic",
+        jdbc: {
+            hostName: "",
+            port: "",
+            databaseName: "",
+            userName: "",
+            password: "",
+            driverClass: "",
+            url: ""
+        },
+        urlStart: "",
+        downloadDriverUrl: ""
+}
+
+export const getSupportedDriversData = function ($scope, RepositoriesRestService, toastr) {
```

Comment:
I think the functions should better be extracted as a service. I wouldn't expect functions in a constants file.

Reply:
In general I strongly agree with Desi here. But other than that, I really don't see what is the point in moving these methods in another file. If these should be extracted, the best IMO is to be created another component with a template and these methods to be in its own controller. As it is now, you generally bind the view controller with a service, which is very bad.